composer/core/evaluator.py

```diff
@@ -51,17 +48,25 @@ def should_eval(state: State, event: Event):
         if eval_at_fit_end and event == Event.FIT_END and state.timestamp.batch != last_batch_seen:
             return True
 
-        if eval_interval.unit == TimeUnit.EPOCH and int(
-                state.timestamp.epoch) % int(eval_interval) == 0 and event == Event.EPOCH_END:
+        assert state.previous_timestamp is not None
+        if eval_interval.unit in {TimeUnit.EPOCH, TimeUnit.BATCH, TimeUnit.TOKEN, TimeUnit.SAMPLE}:
+            previous_count = state.previous_timestamp.get(eval_interval.unit)
+            count = state.timestamp.get(eval_interval.unit)
+        else:
+            assert state.max_duration is not None
+            previous_count = state.previous_timestamp.get(state.max_duration.unit)
+            count = state.timestamp.get(state.max_duration.unit)
```

Comment:
What is this else doing? I'ma bit confused

Reply:
if your eval interval unit is one of the incrementing ones, you track progress in that unit. If your eval interval unit is duration, you track progress with the unit of the max duration instead.

Reply:
can add comment explaining

Reply:
added a comment, please unresolve if you think the comment is unclear or the logic is wrong.